src/app/icd/BUILD.gn

```diff
@@ -32,6 +32,10 @@ source_set("manager-srcs") {
 # All configurations necessary to the ICD Server featureset need to in this configuration
 config("server-config") {
   defines = [ "CHIP_CONFIG_ENABLE_ICD_SERVER" ]
+  if (chip_icd_idle_mode_interval > 0) {
+    defines +=
+        [ "CHIP_CONFIG_ICD_IDLE_MODE_INTERVAL=${chip_icd_idle_mode_interval}" ]
```

Comment:
I have added the getters, but the values are read from the attributes. If the accessor fails, the value defaults to configuration.
Also, I removed the command-line options since now they may conflict with the values from the ZAP configuration.